TestingRandomTests/DiceRollerTests.cs

```diff
@@ -23,22 +24,23 @@ public void Roll_Throws_Format_Exception_For_Malformed_Input_String(string input
             Assert.Throws<FormatException>(Act);
         }
 
-        
         // TODO: Fix this test
-        [Fact]
-        public void Roll_Never_Out_Of_Range()
+        [Theory]
+        [InlineData("2d6", 2, 6)]
+        [InlineData("1d6", 1, 6)]
+        [InlineData("3d6", 3, 6)]
+        public void Roll_Never_Out_Of_Range(string diceExpression, int dieCount, int sideCount)
         {
-            for (var i = 0; i < 100; i++)
-            {
-                var result = _sut.Roll("1d6");
-                if (result < 1 || 6 < result)
-                {
-                    throw new Exception($"Result out of range: {result}");
-                }
-            }
+            var randomMock = new Mock<IRandomizer>();
+            randomMock.Setup(r => r.Next(It.IsAny<int>())).Returns(() => dieCount);
+            var diceRoller = new DiceRoller(randomMock.Object);
+
+            var actual = diceRoller.Roll(diceExpression);
+
+            randomMock.Verify(mock => mock.Next(6), Times.Exactly(dieCount));
```

Comment:
Once `_sut` is set up with a mock, the entire test method can be

    [Theory]
    [InlineData("2d6", 2, 6)]
    [InlineData("3d6", 3, 6)]
    [InlineData("1d8", 1, 8)]
    public void Roll_Never_Out_Of_Range(string diceExpression, int dieCount, int sideCount)
    {
        _ = _sut.Roll(diceExpression);
        
        _randomMock.Verify(r => r.Next(sideCount), Times.Exactly(dieCount));
    }